Tests/MongoSwiftTests/BSONValueTests.swift

```diff
@@ -152,4 +153,64 @@ final class BSONValueTests: MongoSwiftTestCase {
         expect(objectIdFromString.oid).to(equal(oid))
         expect(objectIdFromString.timestamp).to(equal(timestamp))
     }
+
+    /// Test AnyBSONValue Hashable conformance
+    func testHashable() throws {
+        let expected = CodecTests.AllBSONTypes(
+                double: Double(2),
+                string: "hi",
+                doc: ["x": 1],
+                arr: [1, 2],
+                binary: try Binary(base64: "//8=", subtype: .generic),
+                oid: ObjectId(fromString: "507f1f77bcf86cd799439011"),
+                bool: true,
+                date: Date(timeIntervalSinceReferenceDate: 5000),
+                code: CodeWithScope(code: "hi", scope: ["x": 1]),
+                int: 1,
+                ts: Timestamp(timestamp: 1, inc: 2),
+                int32: 5,
+                int64: 6,
+                dec: Decimal128("1.2E+10"),
+                minkey: MinKey(),
+                maxkey: MaxKey(),
+                regex: RegularExpression(pattern: "^abc", options: "imx"))
+
+        let values = Mirror(reflecting: expected).children.map { child in AnyBSONValue(child.value as! BSONValue) }
+        let set = Set<AnyBSONValue>(values)
```

Comment:
let's use a different variable name here because `set` is already a special keyword in swift... for ex. here: https://github.com/mongodb/mongo-swift-driver/blob/b924785a4f0abc8bc4b9d60f8397c52495af0050/Sources/MongoSwift/BSON/Document.swift#L338

while it seems to work ok here, there could be some kind of unintended consequences, and it makes the syntax highlighting funky.

Reply:
fixed